pkg/controller/account/account_controller.go

```diff
@@ -745,38 +745,96 @@ func CreateAccount(reqLogger logr.Logger, client awsclient.Client, accountName,
 	return accountStatus, nil
 }
 
+func checkIAMUserExists(reqLogger logr.Logger, client awsclient.Client, userName string) (bool, error) {
+	// check if username exists for this account
+	_, err := client.GetUser(&iam.GetUserInput{
+		UserName: aws.String(userName),
+	})
+
+	// Retry when getting IAM user information
+	// Sometimes we see a delay before credentials are ready to be user resulting in the AWS API returning 404's
+	attempt := 1
+	for i := 0; i < 10; i++ {
+		attempt++
+		// handle errors
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case iam.ErrCodeNoSuchEntityException:
+					return false, nil
+				case "InvalidClientTokenId":
+					invalidTokenMsg := fmt.Sprintf("Invalid Token error from AWS when attempting get IAM user %s, trying again", userName)
+					reqLogger.Info(invalidTokenMsg)
+					if attempt == 10 {
+						return false, err
+					}
+				case "AccessDenied":
+					checkUserMsg := fmt.Sprintf("AWS Error while checking IAM user %s exists, trying again", userName)
+					controllerutils.LogAwsError(reqLogger, checkUserMsg, nil, err)
+					// We may have bad credentials so return an error if so
+					if attempt == 10 {
+						return false, err
+					}
+				default:
+					controllerutils.LogAwsError(reqLogger, "Unexpected AWS Error when checking IAM user exists", nil, err)
+					return false, err
+				}
+			} else {
+				return false, fmt.Errorf("Unable to check if user %s exists error: %s", userName, err)
+			}
+		}
+		time.Sleep(time.Duration(time.Duration(attempt*5) * time.Second))
```

Comment:
Should this sleep be inside the previous block? The way it is it will sleep even when `err == nil`

Reply:
Also err is never updated, so it will run the switch case with the original value failing every time.